packages/op-tooling/scripts/DisputeGameFactoryPrunner.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import { Hash, GameId, GameType, Timestamp, Claim } from "src/dispute/lib/Types.sol";
+import { IDisputeGame } from "interfaces/dispute/IDisputeGame.sol";
+
+contract DisputeGameFactoryPrunner {
+  event GamePruned(
+    uint256 indexed index,
+    GameId indexed gameId,
+    Hash indexed gameUUID,
+    GameType gameType,
+    address gameProxy
+  );
+
+  uint256[103] internal __gap; // slots 0...102
+  mapping(Hash => GameId) internal _disputeGames; // slot 103
+  GameId[] internal _disputeGameList; // slot 104
+
+  function pruneGames(uint256 _desiredLength) external {
+    require(_desiredLength <= _disputeGameList.length, "Desired length exceeds current length");
+    while (_disputeGameList.length > _desiredLength) {
+      // Retrieve game id to prune
+      uint256 gameIndex_ = _disputeGameList.length - 1;
```

Comment:
Im not sure what is the exact reason why pruning is necessary, but wouldn't it make sense to prune from the start (to prune the oldest games rather than newest ones ) ?